src/Bootstrappers/PrefixCacheTenancyBootstrapper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Illuminate\Cache\Repository;
+use Illuminate\Support\Facades\Cache;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class PrefixCacheTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected null|string $originalPrefix = null;
+    protected string $storeName;
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        $this->originalPrefix = config('cache.prefix');
+        $this->storeName = config('cache.default');
+
+        $this->setCachePrefix('tenant_id_' . $tenant->id);
+    }
+
+    public function revert(): void
+    {
+        if ($this->originalPrefix) {
+            $this->setCachePrefix($this->originalPrefix);
+            $this->originalPrefix = null;
```

Comment:
Why `null` when in `bootstrap()` we use `config('cache.prefix')`? This means that initializing and ending tenancy will break people's central cache.

Also, before fixing this, reproduce this in tests first.

Reply:
I see that there aren't any tests specific to this bootstrapper, just the `cache data is separated` test.

Try to write a proper suite of tests for this bootstrapper. Every other bootstrapper (or its related logic) has them, e.g. just for caching:
- https://github.com/archtechx/tenancy/blob/d6da626f7371b4aa10fab38c61fc6ea2c6065485/tests/CacheManagerTest.php
- https://github.com/archtechx/tenancy/blob/d6da626f7371b4aa10fab38c61fc6ea2c6065485/tests/GlobalCacheTest.php

Reply:
> Why null when in bootstrap() we use config('cache.prefix')? This means that initializing and ending tenancy will break people's central cache.

I tried removing the `null` if check, looks like tests are passing without this.

Reply:
I added test suite https://github.com/archtechx/tenancy/commit/c4f9324aa85d79e94212ddb48fea626784a0653e. 

But to be honest, I don't think we need a separate test suite. Bootstrapper is simple enough. We already tested that it separates data. Most of the tests in this suite are inspired by the CacheManagerTest.

Reply:
Things like this:

> Why null when in bootstrap() we use config('cache.prefix')? This means that initializing and ending tenancy will break people's central cache.

Are what the dedicated test file should cover. Things specific to the **implementation** that we want to make sure work correctly, even in edge cases.

Reply:
https://github.com/archtechx/tenancy/pull/1014/files#diff-77383c3312698f402a6bd8bf0d47ef7622c5f0fe67954a5f055539650025a72d